<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1182</width>
    <height>928</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RTED RuntimeSystem Debugger</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QCodeEditWidget" name="codeEdit1"/>
      <widget class="QCodeEditWidget" name="codeEdit2"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1182</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionEditorSettings"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionResume"/>
    <addaction name="actionSingleStep"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuDebug"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget_4">
   <property name="windowIcon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="windowTitle">
    <string>Memory Locations</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QLineEdit" name="txtMemFilter"/>
     </item>
     <item>
      <widget class="QCheckBox" name="chkShowStack">
       <property name="text">
        <string>Show Stack</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="chkShowHeap">
       <property name="text">
        <string>Show Heap</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTreeView" name="treeMemorySystem">
       <property name="headerHidden">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="editorToolbar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="dockMessages">
   <property name="windowTitle">
    <string>Messages</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QListWidget" name="lstMessages"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="debugToolbar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSingleStep"/>
   <addaction name="actionResume"/>
  </widget>
  <widget class="QDockWidget" name="dockStack">
   <property name="windowIcon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="windowTitle">
    <string>Stack Variables</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLineEdit" name="txtStackFilter">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QTreeView" name="treeStack"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockTypeSystem">
   <property name="windowIcon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="windowTitle">
    <string>Types</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLineEdit" name="txtTypeSystemFilter"/>
     </item>
     <item row="1" column="0">
      <widget class="QTreeView" name="treeTypeSystem"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_5">
   <property name="windowTitle">
    <string>Pointer</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_8">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QLineEdit" name="txtPointerFilter">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTreeView" name="treePointer">
       <property name="headerHidden">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockMemGraphicsView">
   <property name="windowTitle">
    <string>Memory Visualization</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="MemoryGraphicsView" name="memGraphicsView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-save.png</normaloff>:/icons/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-open.png</normaloff>:/icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-save-as.png</normaloff>:/icons/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save as...</string>
   </property>
  </action>
  <action name="actionEditorSettings">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="text">
    <string>Editor Settings...</string>
   </property>
  </action>
  <action name="actionSingleStep">
   <property name="icon">
    <iconset>
     <normaloff>:/redo.png</normaloff>:/redo.png</iconset>
   </property>
   <property name="text">
    <string>Single Step</string>
   </property>
  </action>
  <action name="actionResume">
   <property name="icon">
    <iconset>
     <normaloff>:/debug-resume.png</normaloff>:/debug-resume.png</iconset>
   </property>
   <property name="text">
    <string>Resume</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCodeEditWidget</class>
   <extends>QFrame</extends>
   <header>QCodeEditWidget.h</header>
  </customwidget>
  <customwidget>
   <class>MemoryGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>MemoryGraphicsView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../notSvnTraced/cmakeEnv/widgets/util/util.qrc"/>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
